### DISCUSSION TITLE
Setting names and labels

### DISCUSSION THREAD

**Discussion by nsmith-** [scikit-hep/hist#208] [Q&A] [answered]
How do I set a name or label for an axis of an already created histogram?
I tried
```python
h.axes[0].name = "blah"
```
but that did not work

**Comment by henryiii** [comment, marked as answer]
Labels should be settable that way. Names are read only since they are required to be unique in a histogram and changing them later could break that.

**Reply by nsmith-** [reply]
Would it be possible to create a copy of a histogram with a renamed axis?

**Reply by henryiii** [reply]
You can always do `h_new[...] = h.view(flow=True)` if the axes sizes match.

**Reply by henryiii** [reply]
You can do

```python
ax._ax.metadata["name"] = value
```

If you know what you are doing. Maybe setting them all at once via the AxesTuple could be allowed?

**Reply by nsmith-** [reply]
Thanks, this is really just to get around forgetting to put a name when I first made the histogram. Next time it will have a name so a hacky workaround is fine.

**Reply by Hamidasl** [reply]
![sources.zip](https://github.com/user-attachments/files/17357872/sources.zip)

**Comment by kdlong** [comment]
I have a similar use case: I want to clone an axis while modifying its name. Is the suggested way to do this then

```python
ax_new = copy.deepcopy(ax_old)
ax_new._ax.metadata["name"] = "newname"
``` 

I was looking for something like

```python
ax_new = type(ax_old)(ax_old, name="new name")
```

or some variant but there is no copy constructor (or I don't know how to reproduce the copy constructor behaviour), and I don't particularly want to have separate constructor calls for the different axis types.
